web/src/HUDHeader.tsx

```diff
@@ -0,0 +1,143 @@
+import React, { PureComponent } from "react"
+import styled from "styled-components"
+import * as s from "./style-helpers"
+import { SnapshotHighlight } from "./types"
+import { ReactComponent as SnapshotSvg } from "./assets/svg/snapshot.svg"
+
+type HUDHeaderProps = {
+  podID?: string
+  endpoints?: string[]
+  podStatus?: string
+  showSnapshotButton?: boolean
```

Comment:
i'm surprised these 3 fields (showSnapshotButton, handleOpenModal, highlight) are optional. they seem like ones that you never want to leave off